### PR TITLE
Fix Printer (object types)

### PR DESCRIPTION
**Pull request by Chemaclass**
## 📚 Description

I found a bug in the Printer when I tried to print a struct. The conditional was asking for `AbstractStruct::class === $printerName`, but that's make no sense. The struct is a child of that AbstractStruct, therefore we should check if it's an `instance of`.

## 🔖 Changes

- Fixed the conditional bug for object types.
- Cleanup a bit the `Printer` class
- Fix another minor bug: replace `float` to `double`  
  - Reason: For historical reasons "double" is returned in case of a float, and not simply "float" ([source](https://www.php.net/manual/en/function.gettype.php))

### CODE REVIEW

File: src/php/Printer/Printer.php [hunk 1]
```diff
@@ -53,49 +54,64 @@ private function __construct(bool $readable)
...
             return new PhelArrayPrinter($this);
         }
-        if (AbstractStruct::class === $printerName) {
-            return new StructPrinter($this);
-        }
-        if (Table::class === $printerName) {
+        if ($form instanceof Table) {
             return new TablePrinter($this);
         }
+        if ($form instanceof AbstractStruct) {
```
Comment (jenshaase) [inline src/php/Printer/Printer.php:97]: I think you need to test that before the `Table` since `AbstractStruct` extends `Table`

**Review by jenshaase — CHANGES_REQUESTED** [review]

File: src/php/Printer/Printer.php (hunk 1 above)
Comment (Chemaclass) [reply src/php/Printer/Printer.php:97]: That's why `Table` condition is before and `AbstractStruct` is after.

Comment (jenshaase) [reply src/php/Printer/Printer.php:97]: Yes you are correct!

**Comment by jenshaase** [comment]
It is merged now!

**Pull request merged**